### PR TITLE
fix: stabilize market mobile tab sync OK-56038

### PR DESCRIPTION
**Pull request by limichange**
## Summary
- Stabilizes native mobile Market tab synchronization across watchlist, spot categories, and perps.
- Adds guarded tab-change handling so stale pager events do not overwrite the intended selected tab.

## Intent & Context
This PR packages the current Market tab-sync changes for review against `x`. The implementation focuses on native mobile Market Home behavior where tab presses, programmatic tab jumps, and spot-category state updates need to stay aligned with the persisted selected tab atom.

## Root Cause
The native tab/pager stack can emit delayed or stale `onTabChange` events around programmatic `jumpToTab` calls and direct tab presses. The previous handler accepted every tab-change event, so a stale focused page could push the selected tab/category state back after the user or sync logic had already selected a newer target.

## Design Decisions
- Track an expected tab-change target during direct tab presses and sync-driven jumps, then ignore mismatched tab-change callbacks unless they are backed by a recent user drag.
- Defer early sync jumps briefly on iOS so the native tab press animation can settle before another programmatic jump competes with it.
- Keep pager drag detection platform-local through `pagerProps` instead of weakening tab selection logic globally.

## Changes Detail
- `MarketHomeV2.tsx`: applies pending spot-category selections after category config is available and clears missing pending categories.
- `MobileLayout.native.tsx`: adds expected-target tracking, duplicate/mismatch guards, pager drag detection, and iOS direct tab press animation.
- `useMarketTabsLogic.ts`: avoids redundant selected-tab atom writes.
- `useSyncedMarketTab.ts`: adds pre-jump/defer hooks for resync behavior.

## Risk Assessment
- **Risk Level**: Medium
- **Affected Platforms**: Mobile
- **Risk Areas**: Native Market tab switching, horizontal pager gestures, spot category tabs, watchlist/perps transitions, focused-route resync.

## Test plan
- [x] `yarn lint --fix`
- [x] `yarn lint:staged`
- [x] `yarn tsc:staged`
- [x] `yarn test`
- [ ] Manually verify Market Home tab presses and horizontal swipes on iOS.
- [ ] Manually verify Market Home tab presses and horizontal swipes on Android.

### CODE REVIEW

**Comment by revan-zhang** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/46ed23d3-e6d7-4df8-bc80-5a1f4e1c1ab5?refs=d1741056337c7f92833ed082c2670d39b518e97b&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/46ed23d3-e6d7-4df8-bc80-5a1f4e1c1ab5/license?refs=d1741056337c7f92833ed082c2670d39b518e97b&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: packages/kit/src/views/Market/MarketHomeV2/layouts/MobileLayout.native.tsx [hunk 1]
```diff
@@ -348,15 +438,63 @@ function MobileLayoutComponent({
...
+      ) {
+        return;
+      }
+
+      lastAcceptedTabChangeNameRef.current = tabName;
+      setActiveTabNameRef.current(tabName);
+      handleTabChangeRef.current(tabName);
     },
-    [handleTabChange, setActiveTabName],
+    [clearExpectedTabChangeTarget, currentTabsRef],
```
Comment (sidmorizon) [inline packages/kit/src/views/Market/MarketHomeV2/layouts/MobileLayout.native.tsx:478]: **Complex onTabChangeHandler guard logic — verify behavior with integration testing**

Type: `Flag`

The `onTabChangeHandler` at `MobileLayout.native.tsx:444-478` has three sequential guard conditions that interact with multiple refs (`expectedTabChangeTargetRef`, `lastPagerDraggingAtRef`, `lastAcceptedTabChangeNameRef`, `latestTabStateRef`). I traced through tap, drag, and rapid-tap scenarios and the logic appears correct: (1) rejects events not matching the expected target unless overridden by recent pager drag, (2) rejects events where the pager's focused tab disagrees with the reported tab name, (3) deduplicates repeated events. However, the interaction between these guards, the timer-based expiry of `expectedTabChangeTarget`, and the pager's async events creates a complex state machine that would benefit from integration tests covering edge cases like mid-animation tab taps and drag-then-tap sequences.

Comment (limichange) [reply packages/kit/src/views/Market/MarketHomeV2/layouts/MobileLayout.native.tsx:478]: Thanks for the careful check. I agree this path is stateful, but the guarded behavior is intentional for native PagerView async events. I rechecked the tap, stale onTabChange, drag override, and duplicate-event paths: stale mismatched tab events are dropped, a recent user drag is accepted only when the pager focus already matches the reported tab, and repeated accepted events are deduped here plus by the unchanged-selection guard in useMarketTabsLogic. A full integration test would need RN PagerView/Reanimated timing and is likely to be brittle in Jest, so I am keeping this as native QA/manual-validation coverage for this PR rather than adding a flaky test. Resolving as no code change needed.

**Review by originalix — APPROVED** [review]
